brian2genn/templates/model.cpp

```diff
@@ -194,7 +195,21 @@ void modelDefinition(NNmodel &model)
   {{max_row_length}}
   {% endfor %}
 
-
+  {% for synapses in max_row_length_synapses %}
+  maxRow{{synapses}}= 1;
+  maxCol{{synapses}}= 1;
+  for (size_t Nrow : brian::_dynamic_array_{{synapses}}_N_outgoing) {
```

Comment:
I'm not 100% sure what dynamic arrays actually are but, as I can't see them declared and because this works, I'm guessing they're ``std::vector``, in which case you can do:

```c++
const long maxRow{{synapses}} = std::max_element(brian::_dynamic_array_{{synapses}}_N_outgoing.begin(), brian::_dynamic_array_{{synapses}}_N_outgoing.end());
```
which has the bonus that max row lengths end up const which will provide some safety against it getting messed with by accident.

Reply:
but then, if there are no synapses, maxRowXXX will end up to be 0 and I believe that would cause problems in GeNN?

Reply:
[to confirm: "dynamic arrays" are indeed just `std::vector` in C++ standalone/Brian2GeNN.]

Reply:
If max row length is zero you can skip creating the synapse population entirely though!

Reply:
hum ... that is true ... I went the other way of just taking the max of this and 1 (commit imminent - just waiting for some tests to finish). Performance-wise it probably makes no difference and we don't need the logic to test whether we need to create the population or not. Realistically this is not a common case anyway.

Reply:
That also works - you're totally right that "optimising" out these is likely to be more trouble than it's worth